src/compiler/checker.ts

```diff
@@ -233,6 +233,60 @@ namespace ts {
                 return resolveName(location, escapeLeadingUnderscores(name), meaning, /*nameNotFoundMessage*/ undefined, /*nameArg*/ undefined);
             },
             getJsxNamespace: () => unescapeLeadingUnderscores(getJsxNamespace()),
+
+            isIdenticalTo: (a, b) => checkTypeRelatedTo(a, b, identityRelation, /*errorNode*/ undefined),
+            isSubtypeOf: (a, b) => checkTypeRelatedTo(a, b, subtypeRelation, /*errorNode*/ undefined),
+            isAssignableTo: (a, b) => checkTypeRelatedTo(a, b, assignableRelation, /*errorNode*/ undefined),
+            isComparableTo: areTypesComparable,
+            isInstantiationOf: (a, b) => {
+                return a && b && (a.target === b);
+            },
+
+            lookupGlobalType: name => {

```

Comment:
not sure we should include these..

Reply:
"These" meaning the global variant of the lookup functions? They're primarily convenience, so you don't have to provide the Lib's source file as the location as lookups for globals to the location-specific ones. (And looking at intent, many lint rules wanted to do thing like lookup the global promise or the global array or symbol)